IdentityCore/src/network/response_serializer/MSIDAADResponseSerializer.m

```diff
@@ -31,7 +31,7 @@ - (id)responseObjectForResponse:(NSHTTPURLResponse *)httpResponse data:(NSData *
 {
     NSMutableDictionary *jsonObject = [[super responseObjectForResponse:httpResponse data:data error:error] mutableCopy];
     
-    if (error) { return nil; }
+    if (error && *error) { return nil; }
```

Comment:
technically, we shouldn't expect error to be not nil and change method's behavior based on it. So, it would be better, if we added a local error variable, check for that one to return nil and only check for argument error to assign it.